lib/LedgerSMB/Scripts/setup.pm

```diff
@@ -204,9 +204,7 @@ sub login {
     _init_db($request);
     sanity_checks($database);
     $request->{login_name} = $version_info->{username};
-    # $version_info->{status} isn't always defined by get_info, so useless undefined messages
-    # are generated.
-    if (defined $version_info->{status} && $version_info->{status} eq 'does not exist'){
+    if ($version_info->{status} eq 'does not exist'){
```

Comment:
L284 sets the `appname` key to unknown. I removed the line right after which sets the `exists` key (because we don't advertize it in the API and I think it's a thinko). Then I added back for the entire block where the first connect succeeded, assignment of the `status` key. (The block where I removed the 'exists' is a sub-block of the block which sets the 'status' key.)

Then right here, we check the `status` key, so I think that the check can never use an `undef` value anymore.